### PR TITLE
Wire up nesting relationship delete buttons

### PR DESCRIPTION
**Pull request by laritakr**
Adds functionality described in https://github.com/samvera/hyrax/issues/2295

Add remove buttons to the parent collections and subcollections list on the collection show page. 
Make the buttons functional to remove a parent collection or a subcollection.

With the move of the remove options to the show page, the relationships tab is being removed. It is now unnecessary as all nesting options are available on the show page. (The relationships tab being available only to those with edit authority was problematic, as it didn't give appropriate access to the
relationship management options.)

**==>> Still needs feature specs revised for remove functionality**

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by laritakr** [comment]
Still todo: rework feature specs for collection show page to include remove buttons

File: app/views/hyrax/dashboard/collections/_show_parent_collections.html.erb [hunk 1]
```diff
@@ -1,41 +1,37 @@
...
-  						<strong><%= link_to document, [hyrax, :dashboard, document], id: "src_copy_link_#{document.id}" %></strong>
-  					</li>
-  				</ul>
-  			<% end %>
-  			<button id="show-less-parent-collections" class="btn show-less-parent-collections-btn"><%= t("hyrax.collections.show.show_less_parent_collections") %></button>
-  		</div>
-  	<% end %>
-  <% else %>
-		<% presenter.parent_collections.documents.each do |document| %>
+	<div id="remove_parent_collection">
```
Comment (elrayle) [inline app/views/hyrax/dashboard/collections/_show_parent_collections.html.erb]: What's the thought behind wrapping the parent display section in a div with id remove_parent_collection?

File: app/views/hyrax/dashboard/collections/_subcollection_list.html.erb [hunk 2]
```diff
@@ -1,15 +1,21 @@
 <% if @subcollection_docs.nil? || @subcollection_docs.empty? %>
   <div class="alert alert-warning" role="alert"><%= t('hyrax.collections.show.no_visible_subcollections') %></div>
 <% else %>
-<% @subcollection_docs.each do |document| %>
+<div id="remove_subcollection">
```
Comment (elrayle) [inline app/views/hyrax/dashboard/collections/_subcollection_list.html.erb]: Same here about the display section div including remove in the id.

File: config/locales/hyrax.en.yml [hunk 3]
```diff
@@ -730,8 +734,8 @@ en:
         works:        "Your Works"
       nest_collections_form:
         create_within:          "'%{child_title}' has been added to '%{parent_title}'"
-        create_under:          "'%{child_title}' has been added to '%{parent_title}'"
-        create_under_failed:   "'%{child_title}' cannot be added to '%{parent_title}'"
```
Comment (elrayle) [inline config/locales/hyrax.en.yml]: Did you intend to remove create_under_failed?

File: app/views/hyrax/dashboard/collections/_show_parent_collections.html.erb (hunk 1 above)
Comment (laritakr) [reply app/views/hyrax/dashboard/collections/_show_parent_collections.html.erb]: These were stray tags remaining from some CSS changes I attempted and removed. I will remove them in the next PR.

File: config/locales/hyrax.en.yml (hunk 3 above)
Comment (laritakr) [reply config/locales/hyrax.en.yml]: I found nothing using this translation so i removed it.

**Pull request merged**